src/main/java/scene/Scene.java

```diff
@@ -127,7 +118,21 @@ public final int sceneId() {
      */
     public void addGameObjectToScene(GameObject gameObject) {
         gameObjects.add(gameObject);
-        gameObject.start();
+        if (active) {
+            gameObject.start();
+            addToRenderers(gameObject);
+        }
+    }
+
+    /**
+     * @param gameObject GameObject to be added.
+     */
+    public void removeGameObjectFromScene(GameObject gameObject) {
+        gameObjects.remove(gameObject);
+        this.renderer.remove(gameObject);
+        this.lightmapRenderer.remove(gameObject);
+        this.debugRenderer.remove(gameObject);
+        rendererRegistry.forEach(r -> r.remove(gameObject));
```

Comment:
for cleanliness, you may want to do another function called `removeFromRenderers` which does the same thing as `addToRenderers`

Reply:
Done